### ISSUE TITLE
Building with Mingw Compiler & issues with Mingw-Runtime

### ISSUE THREAD

**Issue by Lucaszw** [wheeler-microfluidics/nadamq#2]
NadaMq can build successfully under the condition that the entire Mingw package is installed in both build and test phases. Attempts to use Mingw-Runtime for test phase without Mingw have failed. Moving forward, it might be best to migrate to MSVC as the default compiler for cython packages.

[Link to Working AppVeyor Build Here](https://ci.appveyor.com/project/Lucaszw/nadamq/build/1.0.42)

bld.bat :
```yaml
@echo off
REM Generate `setup.py` from `pavement.py` definition.
"%PYTHON%" -m paver generate_setup
setlocal enableextensions
md "%PREFIX%"\Library\include\Arduino\NadaMQ
endlocal

REM **Workaround** `conda build` runs a copy of `setup.py` named
REM `conda-build-script.py` with the recipe directory as the only argument.
REM This causes paver to fail, since the recipe directory is not a valid paver
REM task name.
REM
REM We can work around this by wrapping the original contents of `setup.py` in
REM an `if` block to only execute during package installation.
"%PYTHON%" -c "from __future__ import print_function; input_ = open('setup.py', 'r'); data = input_.read(); input_.close(); output_ = open('setup.py', 'w'); output_.write('\n'.join(['import sys', 'import path_helpers as ph', '''if ph.path(sys.argv[0]).name == 'conda-build-script.py':''', '    sys.argv.pop()', 'else:', '\n'.join([('    ' + d) for d in data.splitlines()])])); output_.close(); print(open('setup.py', 'r').read())"

"%PYTHON%" version.py

REM Generate `packet_actions.cpp` from ragel state machine source.
"%PYTHON%" -m paver copy_packet_actions
REM Install source directory as Python package.
"%PYTHON%" -m pip install --no-cache .

if errorlevel 1 exit 1

REM del "%PREFIX%\Scripts\libgcc_s_sjlj-1.dll"
REM del "%PREFIX%\Scripts\libstdc++-6.dll"

REM Run build tests
cd "%PREFIX%"
REM set PATH=%PREFIX%\DLLs;%PATH%
echo "%PATH%"
dir "%SP_DIR%"\\nadamq
dir "%SRC_DIR%"\\nadamq
python -c "import nadamq; import os; print os.getcwd() ; print nadamq.__file__"
nosetests "%SRC_DIR%"\\nadamq\\tests -vv --with-xunit

cp .\\nosetests.xml "%MYCWD%"\\nosetests.xml
dir "%MYCWD%"
REM nosetests -v "%SRC_DIR%"\nadamq\tests
if errorlevel 1 exit 1

REM Copy Arduino library files to Conda Arduino library.
copy "%SP_DIR%"\nadamq\src\Arduino\packet_handler\output_buffer.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\Arduino\packet_handler\packet_handler.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\Arduino\packet_handler\NadaMQ.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\BufferAllocator.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\Packet.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketAllocator.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketHandler.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketParser.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketSocket.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketSocketEvents.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketStream.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\PacketWriter.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\SimpleCommand.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\StreamPacketParser.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\crc-16.c "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\crc-16.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\crc_common.cpp "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\crc_common.h "%PREFIX%"\Library\include\Arduino\NadaMQ
copy "%SP_DIR%"\nadamq\src\packet_actions.cpp "%PREFIX%"\Library\include\Arduino\NadaMQ
if errorlevel 1 exit 1


```

meta.yaml:

```yaml

# source will be downloaded prior to filling in jinja templates
# Example assumes that this folder has setup.py in it
source:
  git_url: ../

package:
  name: nadamq
  version: {{ GIT_DESCRIBE_TAG[1:] }}

build:
  # If this is a new build for the same version, increment the build
  # number. If you do not include this key, it defaults to 0.
  number: {{ GIT_DESCRIBE_NUMBER }}

requirements:
  build:
    - c-array-defs >=0.2.post3
    - cython
    - future
    - libpython # [win]
    - mingw-runtime # [win]
    - mingw # [win]
    - nose
    - numpy
    - path_helpers
    - paver
    - python
    - ragel
    - scons

  run:
    - python
    - c-array-defs >=0.2.post3
    - mingw-runtime # [win]
    - mingw # [win]
    - numpy

test:
  commands:
    - pwd
    - cd "%PREFIX%"
  imports:
    - nadamq
    - nadamq.NadaMq

about:
  home: http://github.com/wheeler-microfluidics/nadamq
  license: MIT

```

AppVeyor.yaml:

```yaml

environment:
  matrix:
    - PYTHON_VERSION: 2.7
      MINICONDA: C:\Miniconda
      PYTHON_ARCH: "32"

version: '1.0.{build}'

init:
  - "ECHO %PYTHON_VERSION% %MINICONDA%"

install:
  - "set PATH=%MINICONDA%;%MINICONDA%\\Scripts;%PATH%"
  - "touch %MINICONDA%\\Lib\\distutils\\distutils.cfg"
  - printf "[build]\ncompiler = mingw32" > %MINICONDA%\Lib\distutils\distutils.cfg
  - conda config --set always_yes yes --set changeps1 no
  - conda config --add channels conda-forge
  - conda config --add channels wheeler-microfluidics
  - conda update -q conda
  - conda create --name nadamq
  - conda install conda-build
  - "call %MINICONDA%\\Scripts\\activate.bat nadamq"
  # - mingw-get update
  # - mingw-get upgrade mingw32-libstdc++
  - conda install --yes anaconda-client nose
  - pip install future
  - FOR /F "tokens=*" %%a in ('conda-build . --output') do SET PACKAGE_LOCATION=%%a
  - echo %PACKAGE_LOCATION%
  - set "MYCWD=%cd%"
  - conda-build .
  # - ps: $package_name = (( $env:PACKAGE_LOCATION -split '\\') | Select-Object -Last 1) -split '\.bz2' | Select-Object -First 1
  # - ps: 7z e $env:PACKAGE_LOCATION -tbzip2
  # - ps: 7z x $package_name -opackage -ttar
  # - ps: $json = Get-Content '.\\package\\info\\index.json' | Out-String | ConvertFrom-Json
  # - ps: $json | Add-Member Foo 'Bar'
  # - ps: $json | ConvertTo-JSON | % { [System.Text.RegularExpressions.Regex]::Unescape($_) } | Set-Content .\package\info\index.json
  # - 7z a package2.tar .\package\* -ttar
  # - 7z a package2.tar.bz2 package2.tar -tbzip2
  # - binstar -t %BINSTAR_TOKEN% upload --force package2.tar.bz2
build: false

test_script:
  - "set PATH=%MINICONDA%;%MINICONDA%\\Scripts;%MINGW_BIN%;%PATH%"
  # - nosetests .\\nadamq\\tests -vv --with-xunit

after_test:
  - ps: $wc = New-Object 'System.Net.WebClient'
  - ps: $wc.UploadFile("https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path $env:MYCWD\\nosetests.xml))

```

**Comment by cfobel** [comment]
I've confirmed this is still a problem as of `v0.12`, 6fd7ead.

```
(C:\Users\chris\Miniconda2\conda-bld\nadamq_1505503657435\_t_env) C:\Users\chris\Miniconda2\conda-bld\nadamq_1505503657435\test_tmp>"C:\Users\chris\Miniconda2\conda-bld\nadamq_1505503657435\_t_env\python.exe" -s "C:\Users\chris\Miniconda2\conda-bld\nadamq_1505503657435\test_tmp\run_test.py"
import: u'nadamq'
import: u'nadamq.NadaMq'
Traceback (most recent call last):
  File "C:\Users\chris\Miniconda2\conda-bld\nadamq_1505503657435\test_tmp\run_test.py", line 5, in <module>
    import nadamq.NadaMq
ImportError: DLL load failed: The specified procedure could not be found.
```

**Comment by cfobel** [comment]
Fixed by `v0.12.1`, f9242eb.
